2-planning/3-documentation.adoc

```diff
@@ -0,0 +1,130 @@
+= Reference Documentation
+
+So, you handed out the mock server to the various API client teams, and thought
+they would get straight to work trying to integrate it. Sadly your inbox is now
+rammed full of questions from those client developers asking how the heck they
+use the API.
+
+*REST devs:* You don't need reference docs, the message is self documenting
+through the rules of HTTP and HATEOAS explains interactions!
+
+*GraphQL devs:* You don't need reference docs, you have GraphiQL!
+
+*gRPC devs:* You don't need reference docs, you have .proto files!
+
+Guess what? If you want that sweet-sweet feedback, you'll need to help your
+clients implement the API, and the best way to avoid inboxmageddon is to
+distribute "API Reference Documentation".
+
+This is usually an overview of methods/resources available in your API, and looks a lot like the sort
+of documentation you'd expect to see for a class or function in any programming
+language. These should always be accompanied by guides and tutorials at a later
+stage, but early on you will want to provide at least the reference
+documentation to folks.
+
+Thankfully you have chosen to write down how it all works with contracts, so
+this will be easy. All we need to do is turn those contract files (probably
+`.json`, `.yaml`, `.proto`, .etc) into some human-readable documentation for
+them to point their faces at. Usually HTML to be thrown on a web server, or
+questionably PDF files for emailing around.
+
+== OpenAPI
+
+OpenAPI has quite a few options for creating reference documentation from source files.
+
+=== Swagger UI
+
+This tool is hard to recommend due to it's very dated appearance.
+
+.A screenshot of Swagger UI, which looks like it was designed by not just a developer, but one who was a big fan of RPC.
+image::images/docs-swagger-ui.png[]
+
+Swagger UI used to be the only option, and it was a major contributor to me
+completely ignoring Swagger for years. It not only looks pretty rough, but it
+hides the data model under a few unclear options, meaning the most important
```

Comment:
> hides the data model under a few unclear options

is that still true? There is a models section on the bottom

![Screenshot_2019-03-19 Swagger Editor](https://user-images.githubusercontent.com/189796/54643012-3702f000-4a96-11e9-9965-ff5e219c46ea.png)

this kind of representation of the data model is missing in redoc.